state/lease/client.go

```diff
@@ -398,19 +301,19 @@ func (client *client) extendLeaseOps(name string, request lease.Request) ([]txn.
 		C:  client.config.Collection,
 		Id: client.leaseDocId(name),
 		Assert: bson.M{
-			fieldLeaseHolder: lastEntry.holder,
-			fieldLeaseExpiry: toInt64(lastEntry.expiry),
-			fieldLeaseWriter: lastEntry.writer,
+			fieldHolder:   lastEntry.holder,
+			fieldStart:    toInt64(lastEntry.start),
+			fieldDuration: lastEntry.duration,
+			fieldWriter:   lastEntry.writer,
 		},
 		Update: bson.M{"$set": bson.M{
-			fieldLeaseExpiry: toInt64(expiry),
-			fieldLeaseWriter: client.config.Id,
+			fieldStart:    toInt64(globalTime),
+			fieldDuration: nextEntry.duration,
```

Comment:
I did this for a potential optimisation.

If you see a lease starts at time T_m, and you've seen that the current global time is T_n where T_n >= T_m, you could potentially use the local clock to assert that the requested duration in *real* time has passed. If you've only got the expiry time, you can't do that safely, because global time can pass slower than real time (i.e. you can work it forwards from the start time, but not backwards from the expiry time).